### ISSUE TITLE
More comprehensible stack traces

### ISSUE THREAD

**Issue by Kacarott** [codewars/runner#116]
Language: In this request I am specifically referring to the `Node 12.x` runner, however I imagine it could easily be extended to many others.

When attempting a kata, debugging errors can be  difficult/annoying. Within the stack traces, the reported filenames (eg. `test.js`) are unhelpful and the reported line numbers are always off (if only a little bit, depending on the amount of preloaded code). The line numbers CAN be helpful but it normally requires extra work to first determine by what margin the line numbers differ, and taking that into account for each debug.

I would propose some form of simple error parsing. Wrap the user solution in a `try catch` block, which would catch any form of error, modify the error (Fix line numbers to what the user sees in their solution, replace file names with more relevant information: Solution, Preloaded, Test Suit, etc.) and then throw the error again, but with the modified text.

This is only a minor enhancement, but I think it would make debugging far more pleasant. Of course there may be factors making this hard/impossible that I haven't considered, in which case feel free to dismiss this.

**Comment by kazk** [comment]
Yes, this is a known issue for _some_ languages. It depends on the language and the language version.

Languages that were added in the early days of Codewars tend to have this issue unless I've already fixed it. Everything is concatenated into a single file.

I've been working on improving this when adding a new language version while trying to minimize the incompatibility. For example, Ruby 3.0 shouldn't have this problem anymore.

For JavaScript, we need to switch to using modules. I have some ideas to do this without breaking every kata, but write some code and need to experiment with actual contents on Codewars.
